src/System.Private.CoreLib/shared/System/Text/Unicode/Utf16Utility.Validation.cs

```diff
@@ -0,0 +1,379 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.X86;
+using System.Numerics;
+using Internal.Runtime.CompilerServices;
+
+#if BIT64
+using nint = System.Int64;
+using nuint = System.UInt64;
+#else // BIT64
+using nint = System.Int32;
+using nuint = System.UInt32;
```

Comment:
Consider something like

```C#
#if DEBUG
static Utf16Utility()
{
    Debug.Assert(sizeof(nint) == IntPtr.Size);
    Debug.Assert(sizeof(nuint) == UIntPtr.Size);
}
```

(just to make sure that the #if here never gets out of sync with something else)